src/main/java/ui/MenuControl.java

```diff
@@ -114,27 +152,31 @@ private void onBoardSaveAs() {
         logger.info("Menu: Boards > Save as");
 
         List<PanelInfo> panelList = panels.getCurrentPanelInfos();
-
-        if (panelList.isEmpty()) {
-            logger.info("Did not save new board");
-            return;
-        }
-
         BoardNameDialog dlg = new BoardNameDialog(prefs, mainStage);
         Optional<String> response = dlg.showAndWait();
-        ui.showMainStage();
-        panels.selectFirstPanel();
-        
+
         if (response.isPresent()) {
             String boardName = response.get().trim();
-            prefs.addBoard(boardName, panelList);
-            prefs.setLastOpenBoard(boardName);
-            ui.triggerEvent(new BoardSavedEvent());
-            logger.info("New board " + boardName + " saved");
+            showFirstPanel(panelList);
+            saveCurrentBoard(panelList, boardName);
             ui.updateTitle();
         }
     }
 
+    private void saveCurrentBoard(List<PanelInfo> panelList, String boardName) {
+        prefs.addBoard(boardName, panelList);
+        prefs.setLastOpenBoard(boardName);
+        ui.triggerEvent(new BoardSavedEvent());
+        logger.info("New board " + boardName + " saved");
+    }
+
+    private void showFirstPanel(List<PanelInfo> panelList) {
+        ui.showMainStage();
+        if (!panelList.isEmpty()) {
+            panels.selectFirstPanel();
```

Comment:
`panelList` originates from `panels`, right? It would be good if this check were done by a `panels` method instead.